config/thisroot.sh

```diff
@@ -39,27 +39,27 @@ clean_environment()
       fi
       if [ -n "${LD_LIBRARY_PATH-}" ]; then
          drop_from_path "$LD_LIBRARY_PATH" "${old_rootsys}/lib"
-         drop_from_path "$LD_LIBRARY_PATH" "${old_rootsys}/lib/root"
+         drop_from_path "$newpath" "${old_rootsys}/lib/root"
          LD_LIBRARY_PATH=$newpath
       fi
       if [ -n "${DYLD_LIBRARY_PATH-}" ]; then
          drop_from_path "$DYLD_LIBRARY_PATH" "${old_rootsys}/lib"
-         drop_from_path "$DYLD_LIBRARY_PATH" "${old_rootsys}/lib/root"
+         drop_from_path "$newpath" "${old_rootsys}/lib/root"
          DYLD_LIBRARY_PATH=$newpath
       fi
       if [ -n "${SHLIB_PATH-}" ]; then
          drop_from_path "$SHLIB_PATH" "${old_rootsys}/lib"
-         drop_from_path "$SHLIB_PATH" "${old_rootsys}/lib/root"
+         drop_from_path "$newpath" "${old_rootsys}/lib/root"
          SHLIB_PATH=$newpath
       fi
       if [ -n "${LIBPATH-}" ]; then
          drop_from_path "$LIBPATH" "${old_rootsys}/lib"
-         drop_from_path "$LIBPATH" "${old_rootsys}/lib/root"
+         drop_from_path "$newpath" "${old_rootsys}/lib/root"
          LIBPATH=$newpath
       fi
       if [ -n "${PYTHONPATH-}" ]; then
          drop_from_path "$PYTHONPATH" "${old_rootsys}/lib"
-         drop_from_path "$PYTHONPATH" "${old_rootsys}/lib/root"
+         drop_from_path "$newpath" "${old_rootsys}/lib/root"
          PYTHONPATH=$newpath
       fi
```

Comment:
Maybe we should also add in 6.36 and 6.38:
```
      # Potential leftovers from < 6.36
      if [ -n "${JUPYTER_CONFIG_DIR-}" ]; then
         drop_from_path "$JUPYTER_CONFIG_DIR" "${old_rootsys}/etc/notebook"
         JUPYTER_CONFIG_DIR=$newpath
      fi
```
as Wile suggested:
https://root-forum.cern.ch/t/the-thisroot-sh-misbahes-in-root-6-34/64513/6?u=ferhue

where as in ROOT 6.32 we should add instead:

```
      # Potential leftovers from future > 6.34
      if [ -n "${JUPYTER_CONFIG_PATH-}" ]; then
         drop_from_path "$JUPYTER_CONFIG_PATH" "${old_rootsys}/etc/notebook"
         JUPYTER_CONFIG_PATH=$newpath
      fi
```